test/utils/describeConformanceV5.js

```diff
@@ -192,6 +193,38 @@ function testThemeComponents(element, getOptions) {
   });
 }
 
+/**
+ * Tests that the root component has the root class
+ * @param {React.ReactElement} element
+ * @param {() => ConformanceOptions} getOptions
+ */
+export function testRootClass(element, getOptions) {
+  it('applies the root class to the root component if it has this class', () => {
+    const { classes, mount } = getOptions();
+    if (classes.root == null) {
+      return;
+    }
+
+    const className = randomStringValue();
+    let wrapper = mount(React.cloneElement(element, { className }));
+
+    // we established that the root component renders the outermost host previously. We immediately
+    // jump to the host component because some components pass the `root` class
+    // to the `classes` prop of the root component.
+    // https://github.com/mui-org/material-ui/blob/f9896bcd129a1209153106296b3d2487547ba205/packages/material-ui/src/OutlinedInput/OutlinedInput.js#L101
+    expect(findOutermostIntrinsic(wrapper).hasClass(classes.root)).to.equal(true);
+    expect(findOutermostIntrinsic(wrapper).hasClass(className)).to.equal(true);
+
+    // Test that classes prop works
+    const classesProp = { ...classes };
```

Comment:
In the future we may need to add option for opting out of this (for example if some component is used as root and `classes` is expected to be passed).